.github/workflows/build.yml

```diff
@@ -0,0 +1,733 @@
+        cxxcommon_version: [ 'v0.1.1' ]
+
+    runs-on: ubuntu-20.04
+    container:
+      image: docker.pkg.github.com/trailofbits/cxx-common/vcpkg-builder-${{ matrix.image.name }}:${{ matrix.image.tag }}
+
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - name: Setup the build paths
+        shell: bash
+        id: build_paths
+        run: |
+          rel_build_path="build"
+          rel_source_path="src"
+          rel_install_path="install"
+          rel_downloads_path="downloads"
+          rel_ccache_path="ccache"
+
+          mkdir -p ${rel_build_path} \
+                   ${rel_install_path} \
+                   ${rel_downloads_path} \
+                   ${rel_ccache_path}
+
+          echo ::set-output name=SOURCE::$(pwd)/${rel_source_path}
+          echo ::set-output name=REL_SOURCE::${rel_source_path}
+          echo ::set-output name=BUILD::$(pwd)/${rel_build_path}
+          echo ::set-output name=REL_BUILD::${rel_build_path}
+          echo ::set-output name=INSTALL::$(pwd)/${rel_install_path}
+          echo ::set-output name=DOWNLOADS::$(pwd)/${rel_downloads_path}
+          echo ::set-output name=CCACHE::$(pwd)/${rel_ccache_path}
+
+      - name: Update the cache (downloads)
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.build_paths.outputs.DOWNLOADS }}
+
+          key: |
+            gitmodules_${{ matrix.image.name }}_${{ matrix.image.tag }}_${{ matrix.llvm }}_${{ github.sha }}
+
+          restore-keys: |
+            gitmodules_${{ matrix.image.name }}_${{ matrix.image.tag }}_${{ matrix.llvm }}
+
+      - name: Update the cache (ccache)
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.build_paths.outputs.CCACHE }}
+
+          key: |
+            gitmodules_${{ matrix.image.name }}_${{ matrix.image.tag }}_${{ matrix.llvm }}_${{ github.sha }}
+
+          restore-keys: |
+            gitmodules_${{ matrix.image.name }}_${{ matrix.image.tag }}_${{ matrix.llvm }}
+
+      - name: Install system dependencies
+        shell: bash
+        run: |
+          apt-get update
+          apt-get install -y pixz \
+                             xz-utils \
+                             make \
+                             ninja-build \
+                             ccache \
+                             rpm
+
+      - name: Clone the anvill repository
+        uses: actions/checkout@v2
+        with:
+          path: ${{ steps.build_paths.outputs.REL_SOURCE }}/anvill
+          fetch-depth: 0
+
+      - name: Clone the remill repository
+        uses: actions/checkout@v2
+        with:
+          repository: 'lifting-bits/remill'
+          path: ${{ steps.build_paths.outputs.REL_SOURCE }}/remill
+          fetch-depth: 0
+
+      - name: Select the remill version
+        run: |
+          remill_commit_id_path="${{ steps.build_paths.outputs.SOURCE }}/anvill/.remill_commit_id"
```

Comment:
The current implementation will just use it if it is found, or print a warning instead. I think we should, for the following reasons:
1. It's useful to know which version of remill we are using when something fails and we want to reproduce
2. Allows us to use git bisect again to look for regression causes
3. Prevents multiple workflow runs against the same anvill commit to behave in different ways due to unpinned dependencies
4. Exactly knowing what we are testing lets us ship things with more confidence as we know whether things are working or not at all times

Reply:
I agree with this. I think a submodule would be ideal, but I know some people don't like submodules. Submodules also have the advantage that dependabot could take care of bumping for us instead of trying to remember to bump manually.